save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt

```diff
@@ -27,13 +36,21 @@ class TestService(private val configProperties: ConfigProperties) {
     }
 
     /**
+     * @param agentId
      * @return Test batches
      */
-    fun getTestBatches(): Mono<List<TestDto>> {
-        val tests = testRepository.retrieveBatches(configProperties.limit, offset.get()).map {
+    @Transactional
+    fun getTestBatches(agentId: String): Mono<List<TestDto>> {
+        val agent = agentRepository.findByContainerId(agentId) ?: error("The specified agent does not exist")
+        log.debug("Agent found: $agent")
+        val execution = agent.execution
+        log.debug("Retrieving tests")
+        val tests = testRepository.retrieveBatches(execution.executionLimit, execution.offset, execution.id!!).map {
             TestDto(it.filePath, it.testSuite.id!!, it.id!!)
         }
-        offset.addAndGet(configProperties.limit)
+        log.debug("Increasing offset of execution - ${agent.execution}")
+        execution.offset += execution.executionLimit
```

Comment:
Let's call it `batchSize`; this name represents its purpose much better and is not bound to DB

Reply:
Fixed

Reply:
I actually meant `executionLimit`, not `offset` :) Offset is fine to keep, but execution limit actually means batch size; the word ``limit` appeared only because this parameter was specific to database queries. And when we will be scaling this parameter. we will be retrieving batch of different size starting from the same offset.